openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+        except fastjsonschema.exceptions.JsonSchemaException as e:
+            raise CheckException(f"JSON does not match schema: {str(e)}")
+
+    def check_collections(self) -> None:
+        c1 = set(self.data.keys())
+        c2 = set(self.models.keys())
+        if c1 != c2:
+            err = "Collections in file do not match with models.py."
+            if c2 - c1:
+                err += f" Missing collections: {', '.join(c2-c1)}."
+            if c1 - c2:
+                err += f" Invalid collections: {', '.join(c1-c2)}."
+            raise CheckException(err)
+
+    def check_model(self, collection: str, model: Dict[str, Any]) -> None:
+        errors = self.check_normal_fields(model, collection)
+
+        if not errors:
+            errors = self.check_template_fields(model, collection)
+
+        if not errors:
+            self.check_types(model, collection)
+            self.check_relations(model, collection)
+
+    def check_normal_fields(self, model: Dict[str, Any], collection: str) -> bool:
+        model_fields = set(
+            x
+            for x in model.keys()
+            if self.is_normal_field(x) or self.is_template_field(x)
+        )
+        collection_fields_types = (
+            x
+            for x in self.models[collection]().get_fields()
+            if self.is_normal_field(x.own_field_name)
+            or self.is_template_field(x.own_field_name)
+        )
+        collection_fields = set(
+            field.own_field_name
+            if not isinstance(field, BaseTemplateField)
+            else field.get_structured_field_name("")
+            for field in collection_fields_types
+        )
+
+        errors = False
+        if collection_fields - model_fields:
+            error = f"{collection}/{model['id']}: Missing fields {', '.join(collection_fields - model_fields)}"
+            self.errors.append(error)
+            errors = True
+        if model_fields - collection_fields:
+            error = f"{collection}/{model['id']}: Invalid fields {', '.join(model_fields - collection_fields)}"
+            self.errors.append(error)
+            errors = True
+        return errors
+
+    def check_template_fields(self, model: Dict[str, Any], collection: str) -> bool:
+        """
+        Only checks that for each replacement a structured field exists and
+        not too many structured fields. Does not check the content.
+        Returns True on errors.
+        """
+        errors = False
+        for template_field in self.get_fields(collection):
+            if not isinstance(template_field, BaseTemplateField):
+                continue
+            field_error = False
+            replacements = model.get(template_field.get_structured_field_name(""))
+
+            if not isinstance(replacements, list):
+                self.errors.append(
+                    f"{collection}/{model['id']}/{template_field.own_field_name}: Replacements for the template field must be a list"
```

Comment:
`own_field_name` contains the name without the `$`, better use `get_template_field_name()`

Reply:
Use get_own_field_name here.